### PR TITLE
build(ingest/hive): Update thrift pin

### PR DESCRIPTION
**Pull request by asikowitz**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/setup.py [hunk 1]
```diff
@@ -206,9 +206,10 @@
     # - 0.6.14 uses pure-sasl instead of sasl so it builds on Python 3.11
     "acryl-pyhive[hive_pure_sasl]==0.6.14",
     # As per https://github.com/datahub-project/datahub/issues/8405
-    # and https://github.com/dropbox/PyHive/issues/417, new versions
-    # of thrift break PyHive's hive+http transport.
-    "thrift<0.14.0",
+    # and https://github.com/dropbox/PyHive/issues/417, version 0.15.0
```
Comment (hsheth2) [inline metadata-ingestion/setup.py:209]:
```suggestion
    # and https://github.com/dropbox/PyHive/issues/417, version 0.14.0
```

**Review by hsheth2 — APPROVED** [review]

**Comment by hsheth2** [comment]
@asikowitz actually looks like they haven't released thrift 0.17.0 yet

https://pypi.org/project/thrift/

**Comment by asikowitz** [comment]
Hm.... on github they're already on 0.19.0 and 0.17.0 came out over a year ago. Kinda unreasonable

File: metadata-ingestion/setup.py [hunk 2]
```diff
@@ -213,11 +213,13 @@
     # - 0.6.12 adds support for Spark Thrift Server
     # - 0.6.13 adds a small fix for Databricks
     # - 0.6.14 uses pure-sasl instead of sasl so it builds on Python 3.11
-    "acryl-pyhive[hive_pure_sasl]==0.6.14",
+    "acryl-pyhive[hive_pure_sasl]==0.6.15",
```
Comment (hsheth2) [inline metadata-ingestion/setup.py:216]:
```suggestion
    # - 0.6.15 adds support for thrift > 0.14 (cherry-picked from https://github.com/apache/thrift/pull/2491)
    "acryl-pyhive[hive_pure_sasl]==0.6.15",
```

**Review by hsheth2 — APPROVED** [review]

**Comment by asikowitz** [comment]
Tests should be fixed by https://github.com/datahub-project/datahub/pull/9254

**Pull request merged**